src/lib/infrastructure/client/controller/browser-list-source-data-controller.ts

```diff
@@ -17,66 +18,7 @@ export default class BrowserListSourceDataController {
     async execute(params: TBrowserListSourceDataControllerParameters): Promise<void> {
 
         try {
-
-            const { researchContextID } = params;
-
-            /**
-             * TODO: move to USECASE
-             */
-            const presenter = new BrowserListSourceDataPresenter(params.response);  // will be injected
-
-            const sourceDataGateway = clientContainer.get<BrowserSourceDataGateway>(GATEWAYS.SOURCE_DATA_GATEWAY);  // will be injected
-
-            let listSourceDataDTO;
-
-            /**
-             * List all source data for client
-             */
-            if (!researchContextID) {
-
-                listSourceDataDTO = await sourceDataGateway.list();
-
-                if (!listSourceDataDTO.success) {
-                    presenter.presentError({
-                        message: listSourceDataDTO.data.message,
-                        operation: "list-source-data",
-                    });
-                    return;
-                }
-
-            /**
-             * List source data for a research context
-             */
-            } else {
-
-                listSourceDataDTO = await sourceDataGateway.listForResearchContext(researchContextID);
-
-                if (!listSourceDataDTO.success) {
-                    presenter.presentError({
-                        message: listSourceDataDTO.data.message,
-                        operation: "list-source-data-for-research-context",
-                    });
-                    return;
-                }
-
-            }
-
-            /**
-             * Logic is the same for both cases after getting the DTO
-             */
-
-            const fileList = listSourceDataDTO.data;
-
-            // NOTE: Need to parse the DTO data to get what we need for the view model
-            // if something in the gateway changes, this will need to be updated
-            const remoteFileList = fileList.filter(
-                (file): file is { type: "remote"; id: string; name: string; relativePath: string; provider: string; createdAt: string; } => file.type === "remote"
-            );
-
-            presenter.presentSuccess({
-                sourceData: remoteFileList,
-            });
-
+            await appRouter.controllers.listSourceData.query({ researchContextID: params.researchContextID } );
```

Comment:
@maany I'm getting an error here saying that the `query` property doesn't exist. Copilot is kinda useless at explaining why this is happening when `query` is very clearly defined on lines 37-66 of app-router.ts... appreciate any help figuring this out!